src/basic/chase-symlinks.c

```diff
@@ -295,23 +298,28 @@ int chase_symlinks_at(
                 }
 
                 /* Otherwise let's see what this is. */
-                child = openat(fd, first, O_CLOEXEC|O_NOFOLLOW|O_PATH);
-                if (child < 0) {
-                        if (errno == ENOENT &&
-                            (flags & CHASE_NONEXISTENT) &&
-                            (isempty(todo) || path_is_safe(todo))) {
-                                /* If CHASE_NONEXISTENT is set, and the path does not exist, then
-                                 * that's OK, return what we got so far. But don't allow this if the
-                                 * remaining path contains "../" or something else weird. */
+                child = r = RET_NERRNO(openat(fd, first, O_CLOEXEC|O_NOFOLLOW|O_PATH));
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return r;
+
+                        if (!isempty(todo) && !path_is_safe(todo))
+                                return r;
 
+                        if (flags & CHASE_NONEXISTENT) {
                                 if (!path_extend(&done, first, todo))
                                         return -ENOMEM;
 
                                 exists = false;
                                 break;
                         }
 
-                        return -errno;
+                        if (!(flags & CHASE_MKDIR_0755))
```

Comment:
So, maybe, we cannot set `CHASE_NONEXISTENT` and `CHASE_MKDIR_0755` simultaneously?
If cannot, then please add an assertion somewhere. If we can, then could you add a brief comment and add testcases for that?

Reply:
It's not allowed, there's a check earlier in the function already

Reply:
Ouch. I guess my eyes were closed.